src/contexts/providers.context.tsx

```diff
@@ -259,15 +287,34 @@ const ProvidersProvider: FC<PropsWithChildren> = (props) => {
     [addNetwork, connectedProvider]
   );
 
+  useEffect(() => {
+    if (connectedProvider.status === "pending") {
+      const web3Provider = getMetamaskProvider();
+      if (web3Provider && env) {
+        void silentlyGetConnectedAccounts(web3Provider).then((accounts) => {
+          const account: string | undefined = accounts[0];
+          if (account) {
+            void connectMetamaskProvider({ env, web3Provider, account });
+          } else {
+            setConnectedProvider({ status: "failed", error: "No connected accounts found" });
+          }
+        });
+      }
+    }
+  }, [connectMetamaskProvider, connectedProvider.status, env]);
+
   useEffect(() => {
     if (account.status === "failed") {
       navigate(routes.login.path);
     }
   }, [account, navigate]);
 
   useEffect(() => {
-    const internalConnectedProvider: Record<string, unknown> | undefined =
-      connectedProvider?.provider.provider;
+    const externalConnectedProvider: Record<string, unknown> | undefined = isAsyncTaskDataAvailable(
```

Comment:
How is the type of this `externalConnectedProvider` variable `Record<string, unknown> | undefined`? Shouldn't this type be `ExternalProvider | undefined`?.

Reply:
I would just use `externalProvider` as the name as it's even more correct in terms of `ethers`.

Reply:
> How is the type of this `externalConnectedProvider` variable `Record<string, unknown> | undefined`? Shouldn't this type be `ExternalProvider | undefined`?.

It's a record because the `ExternalProvider` type does not include the methods `on` and `removeListener`. More on this [here](https://github.com/ethers-io/ethers.js/issues/1396#issuecomment-806380431).

We declare it as a Record so we later can do `"on" in externalProvider && typeof externalProvider.on === "function"` and `"removeListener" in externalProvider && typeof externalProvider.removeListener === "function"` and call the methods.

We could override the type, but this approach is safer than overriding the type and cleaner than parsing the external provider instance, so I think it's ok as it's.